app/app.go

```diff
@@ -123,6 +134,40 @@ func (a *app) WithIoTCore(client iotcore.Client) App {
 	return a
 }
 
+// WithWebhooksAsync enables the asynchronous processing of webhooks requests
+func (a *app) WithWebhooksAsync(async bool) App {
+	a.webhooksAsync = async
+	return a
+}
+
+// WithWebhooksTimeout sets the timeout for webhooks requests
+func (a *app) WithWebhooksTimeout(timeout uint) App {
+	a.webhooksTimeout = time.Duration(timeout * uint(time.Second))
+	return a
+}
+
+func (a *app) ProcessWebhooksQueue(ctx context.Context) {
+	contextDone := ctx.Done()
+	for {
+		select {
+		case <-contextDone:
+			goto quitProcessWebhooksQueue
+		case f := <-a.webhooksQueue:
+			if f == nil {
+				goto quitProcessWebhooksQueue
+			}
+			f()
+		}
+	}
+quitProcessWebhooksQueue:
+	a.webhooksQueueDone <- nil
```

Comment:
For channels as barriers, they should have cap == 0 and be closed upon termination to make it reentrant on the other end:
```suggestion
	close(a.webhooksQueueDone)
```

If more than one go routine in the work queue this must be protected:
```suggestion
	if a.wqMutex.TryLock()
		defer a.wqMutex.Unlock()
		select {
		case <-a.webhooksQueueDone:
		default:
			close(a.webhooksQueueDone)
		}
	}
```

Reply:
I only have one routine, so I don't think I need a lock to protect it.